### PR TITLE
ci: expand build matrix for major OS

### PR DESCRIPTION
**Pull request by Borda**
I suggest including testing on Mac and Win. 
also, the addition of OS builds does not need to be so extensive and could be listed in the `include` section adding 3.8 as the oldest and 3.12 as the latest python version

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -2,12 +2,22 @@ name: Python Fire
...
+    shell: bash
+
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ["3.5", "3.7", "3.8", "3.9", "3.10", "3.11", "3.12"]
+        fail-fast: false
```
Comment (Borda) [inline .github/workflows/build.yml:14]:
```suggestion
        fail-fast: true
```

Comment (hugovk) [reply .github/workflows/build.yml:14]: I recommend `false`, this way you get to see what exactly failed -- is it all versions lower than 3.10? Is it Windows?

With `true`, if just the 3.5/Windows job failed, and all the others were cancelled, you don't get this useful info.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -2,12 +2,22 @@ name: Python Fire
...
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
```
Comment (hugovk) [inline .github/workflows/build.yml:15]: Starting slower ones first can prevent bottlenecks and help builds as a whole finish sooner.

```suggestion
        os: ["windows-latest", "macos-latest", "ubuntu-latest"]
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/build.yml:14]: true is by default, just make it explicit, I normally prefer to run all but it could yield quite extensive testing

**Comment by Borda** [comment]
@dbieber friendly ping if you could have a look... :chipmunk:

**Comment by dbieber** [comment]
I'm open to this!

**Comment by Borda** [comment]
> I'm open to this!

@dbieber updated and resolved conflicts

**Comment by dbieber** [comment]
I'm seeing this error
```
[Invalid workflow file: .github/workflows/build.yml#L18](https://github.com/google/python-fire/actions/runs/10973863694/workflow)
The workflow is not valid. .github/workflows/build.yml (Line: 18, Col: 20): Unexpected value 'false'
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/build.yml:18]:
```suggestion
```

**Comment by Borda** [comment]
> I'm seeing this error

Let's go with @hugovk

**Comment by dbieber** [comment]
> @dbieber if I won't need approval for each run, it would be great (faster)

It should run without approval now. 👍

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -2,18 +2,23 @@ name: Python Fire
...
+    shell: bash
 
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11", "3.12", "3.13.0-rc.2"]
+        os: ["windows-latest", "macos-latest", "ubuntu-latest"]
```
Comment (Borda) [inline .github/workflows/build.yml:18]:
```suggestion
        os: ["macos-latest", "ubuntu-latest"]
```
Let's skip win until the build script is updated

**Pull request merged**